### PR TITLE
Support of googletest using colcon [12009]

### PR DESCRIPTION
**Pull request by richiware**

### CODE REVIEW

File: test/unittest/statistics/rtps/CMakeLists.txt [hunk 1]
```diff
@@ -29,18 +26,20 @@ if(NOT ((MSVC OR MSVC_IDE) AND EPROSIMA_INSTALLER))
...
-        target_include_directories(RTPSStatisticsTests PRIVATE ${GTEST_INCLUDE_DIRS}
+        target_include_directories(RTPSStatisticsTests PRIVATE
             ${PROJECT_SOURCE_DIR}/include
             ${PROJECT_BINARY_DIR}/include
             ${PROJECT_SOURCE_DIR}/src/cpp
             ${Asio_INCLUDE_DIR}
             )
 
-        target_link_libraries(RTPSStatisticsTests fastrtps fastcdr ${GTEST_LIBRARIES} ${GMOCK_LIBRARIES})
+        target_link_libraries(RTPSStatisticsTests fastrtps fastcdr GTest::gtest GTest::gmock)
```
Comment (IkerLuengo) [inline test/unittest/statistics/rtps/CMakeLists.txt:42]:
```suggestion
        target_link_libraries(RTPSStatisticsTests fastrtps fastcdr GTest::gmock)
```
Is not enough with gmock? I see in the rest of cases where both Gtest and Gmock were added, only gmock is added in your changes.

File: test/unittest/rtps/flowcontrol/CMakeLists.txt [hunk 2]
```diff
@@ -44,7 +42,7 @@ if(NOT ((MSVC OR MSVC_IDE) AND EPROSIMA_INSTALLER))
             ${PROJECT_SOURCE_DIR}/include ${PROJECT_BINARY_DIR}/include
             ${PROJECT_SOURCE_DIR}/src/cpp
             )
-        target_link_libraries(ThroughputControllerTests ${GTEST_LIBRARIES} ${GTEST_LIBRARIES} ${GMOCK_LIBRARIES})
+        target_link_libraries(ThroughputControllerTests GTest::gtest GTest::gmock)
```
Comment (IkerLuengo) [inline test/unittest/rtps/flowcontrol/CMakeLists.txt:45]: Same here

**Review by IkerLuengo — APPROVED** [review]
LGTM,

Just a couple of comments, mostly because I'm curious

File: test/unittest/statistics/rtps/CMakeLists.txt (hunk 1 above)
Comment (richiware) [reply test/unittest/statistics/rtps/CMakeLists.txt:42]: I was reading the cmake config file that googletest-distribution generates, and the target `GTest::gmock` imports also `GTest::gtest` and `Threads`

File: test/unittest/rtps/flowcontrol/CMakeLists.txt (hunk 2 above)
Comment (richiware) [reply test/unittest/rtps/flowcontrol/CMakeLists.txt:45]: Here will enough with only `GTest::gmock`.

**Comment by richiware** [comment]
Failed tests not related.

**Pull request merged**